### ISSUE TITLE
Cannot list man pages with apropos

### ISSUE THREAD

**Issue by DamienCassou** [NixOS/nixpkgs#86065]
**Describe the bug**

`apropos` is supposed to return matching man pages but it never finds any.

*TLDR:* It seems that `man` expects a `/run/current-system/sw/share/man/index.bt` file which doesn't exist.

**To Reproduce**

```
$ apropos --regex "^"
^: nothing appropriate.
```

**Expected behavior**

This should list all man pages. The same works fine on debian.

**Additional context**

Here is some debug info:

```
$ apropos --debug --regex "^"
[...]
final search path = /run/current-system/sw/share/man:/home/cassou/.nix-profile/share/man
path=/run/current-system/sw/share/man
warning: can't read the fallback whatis text database /run/current-system/sw/share/man/whatis
path=/home/cassou/.nix-profile/share/man
warning: can't read the fallback whatis text database /home/cassou/.nix-profile/share/man/whatis
^: nothing appropriate.
```

There are plenty of man pages in my system though:

```
$ ls /run/current-system/sw/share/man/
cs  da  de  fi  fr  hu  id  it  ja  ko  man1  man3  man4  man5  man7  man8  pl  pt_BR  ru  sv  tr  zh_CN  zh_TW

$ ls /run/current-system/sw/share/man/man1 | wc --lines1373
```

Some debugging in the C code reveals that `man` expects a `/run/current-system/sw/share/man/index.bt` file which doesn't exist.

**Notify maintainers**

/cc @ryantm @matthewbauer 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.33, NixOS, 20.09.git.87aea649013 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(cassou): `""`
 - channels(root): `"nixos-20.09pre222244.22a3bf9fb9e"`
 - nixpkgs: `/home/cassou/Documents/projects/nixos/system`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by eadwu** [comment]
https://github.com/NixOS/nixpkgs/pull/76841
Though I personally use https://github.com/eadwu/flakes/blob/custom/modules/apropos.nix which addresses some of the comments pointed in the PR, though not sure how effective they are at it.

**Comment by DamienCassou** [comment]
Thank you @eadwu.

Closing because it is a duplicate of https://github.com/NixOS/nixpkgs/issues/14472.
